relay-server/src/utils/rate_limits.rs

```diff
@@ -103,7 +103,7 @@ fn infer_event_category(item: &Item) -> Option<DataCategory> {
         ItemType::MetricBuckets => None,
         ItemType::FormData => None,
         ItemType::UserReport => None,
-        ItemType::Profile => None,
+        ItemType::Profile => Some(DataCategory::Profile),
```

Comment:
Profiles aren't events, so `infer_event_category` should not return something here.